src/modules/ObservableArrayProxy.ts

```diff
@@ -0,0 +1,33 @@
+import { ObservableArray as KnockoutObservableArray } from 'knockout';
+
+export default class ObservableArrayProxy<T> extends Array<T> implements Array<T> {
```

Comment:
https://github.com/Aegyo/incremental-game-template/commit/6637be411be315976bdcaa436fca6ad9c062fd73

A request from Isha a while ago, but I think he ditched knockout entirely for that project

Reply:
I'd appreciate that. I spent a few hours trying to come up with a good solution, but kept hitting roadblocks with the fact that we are tricking TS into believing it is something that it isn't. The `Array<T>` lets us call array methods on it, even though it is a proxied observable, and the proxy passes those through to the underlying array. After letting it stew for a bit, I think the best approach is to clean up the places that are calling it (player constructor looping over the `mineInventory` observable as if it were an array, instead of getting the observable's value to loop over).

Reply:
Similar to what we did in #948, I've removed everything related to `ObservableArrayProxy` and replaced it with direct use of `ko.observableArray`

Reply:
Yep I did, happy to discuss why :)